libraries/AP_HAL_ChibiOS/system.cpp

```diff
@@ -203,6 +204,9 @@ void panic(const char *errormsg, ...)
     vprintf(errormsg, ap);
     va_end(ap);
 
+    gcs().send_text(MAV_SEVERITY_ALERT, errormsg, ap);
+    hal.scheduler->delay(1000);  // Wait for the time when the message will be sent
+
     hal.scheduler->delay_microseconds(10000);
     while (1) {
         vprintf(errormsg, ap);
```

Comment:
I don't suppose you could do some research here, could you?

We're using "ap" after we've done a `va_end` on it - that seems rather wrong to me, and probably invalid.  Could you verify that, perhaps?  Note that your new code is also using it.

Reply:
I've now done this research, and it is *not* valid to re-use ap after va_end has been called on it.

I've created https://github.com/ArduPilot/ardupilot/pull/16223 to fix this pre-existing problem, but your patch here has the same issue.